src/data/set/equitable.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2021 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import algebra.big_operators.order
+import data.nat.basic
+
+/-!
+# Equitable functions
+
+This file defines equitable functions.
+
+A function `f` is equitable on a set `s` if `f a₁ ≤ f a₂ + 1` for all `a₁, a₂ ∈ s`. This is mostly
+useful when the domain of `f` is `ℕ` or `ℤ` (or more generally a successor order).
+
+## TODO
+
+`ℕ` can be replaced by any `succ_order` + `conditionally_complete_monoid`, but we don't have either
+of those yet.
+-/
+
+open_locale big_operators
+
+variables {α β : Type*}
+
+namespace set
+
+/-- A set is equitable if no element value is more than one bigger than another. -/
+def equitable_on [has_le β] [has_add β] [has_one β] (s : set α) (f : α → β) : Prop :=
```

Comment:
Does it make sense to also have `equitable` which means `equitable_on univ`? Like we have `continuous(_on)` etc...

Reply:
The definition is already so contrived that I doubt it'll ever be useful, but I sure I can add it.

Reply:
Ok, if this is only used in your formalisation of Szemeredi, and you don't have other applications in mind, let's leave it like this.